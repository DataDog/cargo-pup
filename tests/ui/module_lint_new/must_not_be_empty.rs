// This product includes software developed at Datadog (https://www.datadoghq.com/) Copyright 2024 Datadog, Inc.

//@compile-flags: --crate-name test_must_not_be_empty
//@compile-flags: --crate-type lib

// Non-empty module (allowed)
pub mod non_empty_module {
    pub fn valid_function() {
        println!("This module has content so it's valid");
    }
    
    pub const VALID_CONST: &str = "Also valid content";
}

// Empty module (should trigger an error)
pub mod empty_module { //~ ERROR: Module must not be empty
}

// Another empty module (should also trigger an error)
pub mod another_empty_module { //~ ERROR: Module must not be empty
} 
