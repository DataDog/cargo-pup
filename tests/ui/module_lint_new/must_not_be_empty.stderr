error: Module must not be empty
  --> tests/ui/module_lint_new/must_not_be_empty.rs:14:1
   |
LL | / pub mod empty_module {
LL | | }
   | |_^
   |
   = help: Add content to this module or remove it
   = note: Applied by cargo-pup rule 'must_not_be_empty_test'.
   = note: `#[deny(module_must_not_be_empty_lint)]` on by default

error: Module must not be empty
  --> tests/ui/module_lint_new/must_not_be_empty.rs:18:1
   |
LL | / pub mod another_empty_module {
LL | | } 
   | |_^
   |
   = help: Add content to this module or remove it
   = note: Applied by cargo-pup rule 'must_not_be_empty_test'.

error: aborting due to 2 previous errors

